lib/streamlit/util.py

```diff
@@ -258,12 +258,14 @@ def open_browser(url):
 
     system = platform.system()
 
-    if system == "Windows":
+    if (system == "Windows"
+        or system == "Linux" and not _is_executable_in_path("xdg-open")):
```

Comment:
Put paretheses around the Linux part of the `if` so it's clearer that `_is_executable_in_path` is only for Linux.

Also, if you feel like modifying the code a little more, it would actually be nice to move this whole logic inside the `if Linux` block below. This way the code would have a cleaner hierarchy: check OS first, then do something os-specific.

```py
if IS_WINDOWS:
  # Windows code
elif IS_LINUX:
  # Linux code
elif IS_MAC:
  # Mac code
else:
  # Other code.
```